pkg/k8s/service.go

```diff
@@ -444,11 +444,26 @@ func NewService(ips []net.IP, externalIPs, loadBalancerIPs, loadBalancerSourceRa
 		loadBalancerSourceCIDRs[cidr.String()] = cidr
 	}
 
+	// If EnableNodePort is not true we do not want to process
+	// events which only differ in external or load balancer IPs.
+	// By omitting these IPs in the returned Service object, they
+	// are no longer considered in equality checks and thus save
+	// CPU cycles processing events Cilium will not act upon.
 	if option.Config.EnableNodePort {
 		k8sExternalIPs = parseIPs(externalIPs)
 		k8sLoadBalancerIPs = parseIPs(loadBalancerIPs)
 	}
 
+	// The BGP LB Announcement feature require that
+	// loadBalancerIPs be parsed. This is because
+	// an event must occur when a Service's Status field
+	// is updated with a new Ingress, ultimately triggering a
+	// BGP announcement. If we do not parse loadBalancerIPs
+	// this will not occur.
+	if option.Config.BGPAnnounceLBIP && k8sLoadBalancerIPs == nil {
```

Comment:
Small nit, this can be done as an `else if` where it's the `else` of `if option.Config.EnableNodePort` and then we don't need to double check `k8sLoadBalancerIPs == nil`.

Reply:
Updated.